centreon/src/Core/HostTemplate/Domain/Model/NewHostTemplate.php

```diff
@@ -0,0 +1,387 @@
+     * @param string $iconAlternative
+     * @param string $comment
+     * @param bool $isActivated
+     * @param bool $isLocked
+     *
+     * @throws AssertionFailedException
+     */
+    public function __construct(
+        protected string $name,
+        protected string $alias,
+        protected ?SnmpVersion $snmpVersion = null,
+        protected string $snmpCommunity = '',
+        protected ?int $timezoneId = null,
+        protected ?int $severityId = null,
+        protected ?int $checkCommandId = null,
+        protected string $checkCommandArgs = '',
+        protected ?int $checkTimeperiodId = null,
+        protected ?int $maxCheckAttempts = null,
+        protected ?int $normalCheckInterval = null,
+        protected ?int $retryCheckInterval = null,
+        protected YesNoDefault $activeCheckEnabled = YesNoDefault::Default,
+        protected YesNoDefault $passiveCheckEnabled = YesNoDefault::Default,
+        protected YesNoDefault $notificationEnabled = YesNoDefault::Default,
+        protected array $notificationOptions = [],
+        protected ?int $notificationInterval = null,
+        protected ?int $notificationTimeperiodId = null,
+        protected bool $addInheritedContactGroup = false,
+        protected bool $addInheritedContact = false,
+        protected ?int $firstNotificationDelay = null,
+        protected ?int $recoveryNotificationDelay = null,
+        protected ?int $acknowledgementTimeout = null,
+        protected YesNoDefault $freshnessChecked = YesNoDefault::Default,
+        protected ?int $freshnessThreshold = null,
+        protected YesNoDefault $flapDetectionEnabled = YesNoDefault::Default,
+        protected ?int $lowFlapThreshold = null,
+        protected ?int $highFlapThreshold = null,
+        protected YesNoDefault $eventHandlerEnabled = YesNoDefault::Default,
+        protected ?int $eventHandlerCommandId = null,
+        protected string $eventHandlerCommandArgs = '',
+        protected string $noteUrl = '',
+        protected string $note = '',
+        protected string $actionUrl = '',
+        protected ?int $iconId = null,
+        protected string $iconAlternative = '',
+        protected string $comment = '',
+        protected bool $isActivated = true,
+        protected readonly bool $isLocked = false
+    ) {
+        $shortName = (new \ReflectionClass($this))->getShortName();
+
+        // Assertions on string properties
+        $this->name = trim($name);
+        $this->alias = trim($alias);
+        $this->snmpCommunity = trim($snmpCommunity);
+        $this->checkCommandArgs = trim($checkCommandArgs);
+        $this->eventHandlerCommandArgs = trim($eventHandlerCommandArgs);
+        $this->noteUrl = trim($noteUrl);
+        $this->note = trim($note);
+        $this->actionUrl = trim($actionUrl);
+        $this->iconAlternative = trim($iconAlternative);
+        $this->comment = trim($comment);
+
+        Assertion::notEmptyString($name, "{$shortName}::name");
```

Comment:
The trim function was applied to the protected property, not to the "name" argument.